nc-news-bh-final-vite/src/components/NewCommentForm.jsx

```diff
@@ -0,0 +1,40 @@
+import {useState, useEffect} from 'react';
+import {useParams} from 'react-router-dom'
+import {postComment} from '../utils/api.js';
+
+
+const NewCommentForm = ({setNewComments})=>{
+    const {article_id} = useParams();
+
+    const [newComment, setNewComment] = useState({
+        username: "",
+        body: ""
+    });
+
+    function handleSubmit(e){
+        e.preventDefault();
+        postComment(newComment, article_id).then((newComment)=>{
+            setNewComments((currComments)=> [newComment, ...currComments]) 
+            setNewComment({
+                username: "",
+                body: ""
+            })
+        })     
+    }
+
+    return(
+        <form id='comment-form' onSubmit={handleSubmit}>
+        <fieldset>
+        <legend>Post a comment</legend>
+        <label htmlFor="username">Enter your name: </label>
```

Comment:
This works, but eventually it should be changed so that we don’t have to enter the username here and it just picks up whoever is logged in via useContext.